models/staking/ethereum/staking_ethereum_deposits.sql

```diff
@@ -13,56 +14,56 @@
 WITH deposit_events AS (
     SELECT d.evt_block_time AS block_time
     , d.evt_block_number AS block_number
-    , CAST(d.evt_index AS VARCHAR(100)) AS evt_index
+    , d.evt_index
     , d.evt_tx_hash AS tx_hash
-    , CAST(bytea2numeric_v3('0x' || SUBSTR(d.amount, 15, 2) || SUBSTR(d.amount, 13, 2) || SUBSTR(d.amount, 11, 2) || SUBSTR(d.amount, 9, 2) ||
-        SUBSTR(d.amount, 7, 2) || SUBSTR(d.amount, 5, 2) || SUBSTR(d.amount, 3, 2))/POWER(10, 9) AS DOUBLE) AS amount
+    , from_big_endian_64(reverse(d.amount)) / POWER(10, 9) AS amount
     , d.contract_address
-    , CAST(bytea2numeric_v3('0x' || SUBSTR(d.index, 15, 2) || SUBSTR(d.index, 13, 2) || SUBSTR(d.index, 11, 2) || SUBSTR(d.index, 9, 2) ||
-        SUBSTR(d.index, 7, 2) || SUBSTR(d.index, 5, 2) || SUBSTR(d.index, 3, 2)) AS DECIMAL(38,0)) AS deposit_index
+    , from_big_endian_64(reverse(d.index)) AS deposit_index
```

Comment:
tbh, i don't really know what this function does 🤷‍♂️  but do maybe we want to use one of the dunesql functions and output to `uint256` for amount? like [this](https://dune.com/docs/query/DuneSQL-reference/Functions-and-operators/varbinary/#bytearray_to_uint256)?

Reply:
Oh didn't realise this was a Spark function. Big endian reverses the order of the bytes which is necessary to decode the values here. I don't see any endian (big or little) functions available [here](https://dune.com/docs/query/DuneSQL-reference/Functions-and-operators/varbinary/#bytearray_to_uint256). Or maybe those could be added through UDFs? @andrewhong5297 would that be possible?

Reply:
does [bytearray_reverse()](bytearray_reverse) not work? i know i've seen someone using reverse logic in spellbook 🤔

Reply:
like [here](https://github.com/duneanalytics/spellbook/blob/a4c10784d3ef22d88209977b0196853a27ca6bf5/macros/models/_sector/nft/enrich_trades.sql#L119) from @0xRobin -- maybe rob can help here? sorry, i haven't had a chance to pull this into a query and look at results myself

Reply:
Thank you I figured it out, `bytearray_to_bigint(reverse())` works perfectly!